### PR TITLE
Migrate etcd data using correct etcd version in case of previous crash

### LINKED ISSUE #41323: etcd migration tool crashloops if there is a problems during first attempt
If in line line #190 is executed also for step 2.2.1 if the tool failed during first run of upgrade. I guess we should also check for the step and compare it to version.

@mml

### LINKED ISSUE #41324: Add more logs to etcd migration tool
At the very least it should be clear when (timestamp) the migration started and finished.

@mml

### PR DESCRIPTION
**Pull request by wojtek-t**
Fix #41324
Fix #41323

@mml

### CODE REVIEW

File: cluster/images/etcd/migrate-if-needed.sh [hunk 1]
```diff
@@ -52,6 +52,8 @@ if [ -z "${DATA_DIRECTORY:-}" ]; then
   exit 1
 fi
 
+echo "Detecting if migration is needed"
```
Comment (fgrzadkowski) [inline cluster/images/etcd/migrate-if-needed.sh]: We also need time here to see how long it took.

Comment (wojtek-t) [inline cluster/images/etcd/migrate-if-needed.sh]: Done

**Comment by wojtek-t** [comment]
@fgrzadkowski - PTAL

**Comment by wojtek-t** [comment]
Fixed bad quotations in the script around dates. Reapplying label.

**Pull request merged**